ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -610,7 +624,13 @@ private module ParameterNodes {
         pos.isHashSplat()
         or
         parameter = callable.getParameter(0).(SplatParameter) and
+        not exists(callable.getParameter(1)) and
         pos.isSplatAll()
+        or
+        exists(int n | n > 0 |
+          parameter = callable.getParameter(n).(SplatParameter) and
+          pos.isSplat(n)
```

Comment:
Don't we need `not exists(callable.getParameter(n+1))` here? Otherwise, I would think that in
```rb
def foo(a, *splats, b)
end

foo(0, *[1, 2])
```
both `1` and `2` get passed into the splat parameter.